### PR TITLE
Fix empty list handling in maxpool | fix(atenlib)

### PR DESCRIPTION
**Pull request by justinchuby**
Previously it was passed in as an empty list and graph_builder fails when the list is empty.

Discovered in https://github.com/pytorch/pytorch/pull/99284

- Added maxpool_2d/3d custom OpInfo tests. Skipping 3d for now due to shape mismatch.
- Pin torch-nightly version to a requirements file

### CODE REVIEW

**Review by BowenBao — COMMENTED** [review]
Should we add extra custom opinfo for coverage, or contribute to pt opinfo?

**Comment by justinchuby** [comment]
We should. An in this case it looks like the OpInfo for torch and the aten signature doesn't match very well.

**Comment by titaiwangms** [comment]
> We should. And in this case it looks like the OpInfo for torch and the aten signature doesn't match very well.

We should add this case into our own OpInfo test.

**Comment by justinchuby** [comment]
> > We should. And in this case it looks like the OpInfo for torch and the aten signature doesn't match very well.
> 
> We should add this case into our own OpInfo test.

Done

**Review by titaiwangms — APPROVED** [review]
LGTM

**Comment by titaiwangms** [comment]
I wonder if there is a way that exporter can also use extra_Opinfo

**Comment by justinchuby** [comment]
> I wonder if there is a way that exporter can also use extra_Opinfo

I can think of coping this there. Any suggestions?

**Comment by justinchuby** [comment]
Somethings off with convolution

File: onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py [hunk 1]
```diff
@@ -799,6 +801,11 @@ def _where_input_wrangler(
         reason="fixme: current implementation gets shape inference error",
         test_class_name="TestOutputConsistencyFullGraph",
     ),
+    xfail(
```
Comment (fatcat-z) [inline onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py:804]: Is this a bug?

Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py:804]: It is. A bug I haven't figured out yet

File: onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py [hunk 2]
```diff
@@ -701,6 +701,8 @@ def _where_input_wrangler(
     "index_select": core_ops.aten_index_select,
     "layer_norm": core_ops.aten_layer_norm,
     "max": core_ops.aten_max,
+    "max_pool2d": nn_ops.aten_max_pool2d,  # Custom from extra_opinfo
```
Comment (fatcat-z) [inline onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py:704]: We've already had nn.functional.max_pool2d and nn.functional.max_pool3d tests. If we want to use this custom testing to replace them, let's remove those 2 ones from the test file.

File: onnxscript/tests/function_libs/torch_aten/extra_opinfo.py [hunk 3]
```diff
@@ -201,6 +202,95 @@ def sample_inputs_layer_norm(op_info, device, dtype, requires_grad, **kwargs):
         )
 
 
+class _TestParamsMaxPoolEmptyStrideBase:
```
Comment (fatcat-z) [inline onnxscript/tests/function_libs/torch_aten/extra_opinfo.py:205]: Why do we keep a local custom test data here?

File: onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py (hunk 1 above)
Comment (fatcat-z) [reply onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py:804]: > I wonder if there is a way that exporter can also use extra_Opinfo

What's the scenario of using these op information during exporting?

File: onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py (hunk 2 above)
Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py:704]: The new tests just test for when stride is empty

File: onnxscript/tests/function_libs/torch_aten/extra_opinfo.py (hunk 3 above)
Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_aten/extra_opinfo.py:205]: I just copied the class and edited it to contain empty strides. Any suggestions?

File: onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py:804]: We can probably enrich the pytorch opinfo db @titaiwangms

**Pull request merged**